### PR TITLE
Improve `tsh play` UX

### PR DESCRIPTION
**Pull request by mcbattirola**
This PR includes two changes to improve the UX of `tsh play`, corresponding to the following issues:

- https://github.com/gravitational/teleport/issues/11207
- gravitational/cloud#1351

## tsh play shows a confusing error for desktop sessions

Old output:

```
$ tsh play 9e9d708b-6c4e-4bfa-bbdf-e19d464fbad4
ERROR: offset 0 not found for session 9e9d708b-6c4e-4bfa-bbdf-e19d464fbad4
```

With this PR, this now outputs:

```
$ ./build/tsh play 9e9d708b-6c4e-4bfa-bbdf-e19d464fbad4
ERROR: Desktop sessions cannot be viewed with tsh. Please use the browser to play the sessions. Click on the URL to view the session in the browser:
[proxy address]/web/cluster/[cluster name]/session/9e9d708b-6c4e-4bfa-bbdf-e19d464fbad4?recordingType=desktop&durationMs=42562
```

## tsh play session failures outputs session bucket info

This PR changes the error message when trying to play a session recording with `tsh play <GUID>` that doesn't exits from:

```bash
ERROR: [s3 bucket]/7ecdcc48-e9e0-4d6e-9af8-c0d26a126888.tar not found
```

To

```bash
ERROR: Recoding for session 7ecdcc48-e9e0-4d6e-9af8-c0d26a126888 not found.
```

The reason is to improve the user experience, as stated in https://github.com/gravitational/cloud/issues/1351.

As you can see in the issue's comments, it is still possible to see the session record bucket name for cloud users on the logs, so there's no need to hide it here. The objective is just to make the message clear and hide unnecessary details.
___

Note that this doesn't change the error message when trying to `play` from a local file:

```bash
$  ./build/tsh play 7ecdcc48-e9e0-4d6e-9af8-c0d26a126888.tar
ERROR: open 7ecdcc48-e9e0-4d6e-9af8-c0d26a126888.tar: no such file or directory
```

### Implementation Details

I didn't `trace.Wrap` the error into a new one because this would cause both to be displayed to the user:

```bash
ERROR: Recoding for session 7ecdcc48-e9e0-4d6e-9af8-c0d26a126888 not found.
[s3 bucket]/7ecdcc48-e9e0-4d6e-9af8-c0d26a126888.tar not found
```

Since this makes us lose the original error, I added `log.WithError(err).Debug("...")`, so users running it with `-d` will still get to see the original error trace:

```bash
DEBU [TSH]       error playing session error:[
ERROR REPORT:
Original Error: *trace.NotFoundError [s3 bucket]/7ecdcc48-e9e0-4d6e-9af8-c0d26a126888.tar not found
Stack Trace:

Caught:
        [original error stack trace]

ERROR REPORT:
Original Error: *trace.NotFoundError Recoding for session 7ecdcc48-e9e0-4d6e-9af8-c0d26a126888 not found.
Stack Trace:
        [new error stack trace]
User Message: Recoding for session 7ecdcc48-e9e0-4d6e-9af8-c0d26a126888 not found.
```

Closes gravitational/cloud#1351

### CODE REVIEW

**Comment by zmb3** [comment]
I wonder if we could also include https://github.com/gravitational/teleport/issues/11207 in scope while you're here, as it's closely related. What do you think?

**Comment by mcbattirola** [comment]
> I wonder if we could also include #11207 in scope while you're here, as it's closely related. What do you think?

Makes sense to me, I'll convert this back to draft and add this. Thanks!

**Comment by zmb3** [comment]
Thanks @mcbattirola!

The trick here will just be identifying whether you get a `SessionEnd` event or a `WindowsDesktopSessionEnd` event.

File: tool/tsh/tsh.go [hunk 1]
```diff
@@ -1236,6 +1236,10 @@ func playSession(cf *CLIConf) error {
 		return trace.Wrap(err)
 	}
 	if err := tc.Play(cf.Context, cf.Namespace, cf.SessionID); err != nil {
+		if trace.IsNotFound(err) {
+			log.WithError(err).Debug("error playing session")
+			return trace.NotFound("Recoding for session %s not found.", cf.SessionID)
```
Comment (alistanis) [inline tool/tsh/tsh.go:1241]:
```suggestion
			return trace.NotFound("Recording for session %s not found.", cf.SessionID)
```

File: lib/client/api.go [hunk 2]
```diff
@@ -2060,6 +2060,17 @@ func (tc *TeleportClient) Play(ctx context.Context, namespace, sessionID string)
 		return trace.Wrap(err)
 	}
 
+	// Return an error if it is a desktop session
+	if len(sessionEvents) > 0 {
+		if sessionEvents[0].GetType() == events.WindowsDesktopSessionStartEvent {
+			url := getDesktopEventWebURL(tc.localAgent.proxyHost, proxyClient.siteName, sid, sessionEvents)
+			message := "Desktop sessions cannot be viewed with tsh." +
+				" Please use the browser to play the sessions." +
```
Comment (zmb3) [inline lib/client/api.go:2068]:
```suggestion
				" Please use the browser to play this session." +
```

**Review by zmb3 — APPROVED** [review]
Thanks!

File: lib/client/api.go [hunk 3]
```diff
@@ -4308,3 +4319,16 @@ func findActiveDatabases(key *Key) ([]tlsca.RouteToDatabase, error) {
 	}
 	return databases, nil
 }
+
+// getDesktopEventWebURL returns the web UI URL users can access to
+// watch a desktop session recording in the browser
+func getDesktopEventWebURL(proxyHost string, cluster string, sid *session.ID, events []events.EventFields) string {
+	if events == nil || len(events) < 1 {
+		return ""
+	}
```
Comment (GavinFrazar) [inline lib/client/api.go:4328]:
```suggestion
func getDesktopEventWebURL(proxyHost string, cluster string, sid *session.ID, events []events.EventFields) string {
	if len(events) < 1 {
		return ""
	}
```
nit: you can just use the `len` nil slices in go

**Review by GavinFrazar — APPROVED** [review]

**Pull request merged**